toolchain/lower/file_context.cpp

```diff
@@ -372,6 +620,24 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id,
   Mangler m(*this);
   std::string mangled_name = m.Mangle(function_id, specific_id);
 
+  // Create a unique fingerprint for the function type
+  // For now, compute the function type fingerprint only for specifics, though
+  // we might need it for all functions in order to create a canonical
+  // fingerprint across translation units.
+  if (specific_id.has_value()) {
+    llvm::BLAKE3 function_type_fingerprint;
+    RawStringOstream os;
+    if (function_type_info.type) {
+      function_type_info.type->print(os);
+    }
```

Comment:
Is the function type always non-null? Maybe (with `\n` separator):

```suggestion
    function_type_info.type->print(os);
```